tests/helpers/video_utils.py

```diff
@@ -51,28 +51,28 @@ def save_control_data_from_video(path_to_video: str, name: str) -> None:
 
     See Also
     --------
+
     tests/utils/video_tester.py : read control data and compare with output of test
     """
-    path_to_sections = os.path.join(
-        pathlib.Path(path_to_video).parent.absolute(), "sections"
-    )
-    tests_directory = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-    path_control_data = os.path.join(tests_directory, "control_data", "videos_data")
+    orig_path_to_sections = Path(path_to_video)
+    path_to_sections = orig_path_to_sections.parent.absolute().joinpath("sections")
+    tests_directory = Path(__file__).absolute().parent.parent
+    path_control_data = Path(tests_directory).joinpath("control_data", "videos_data")
     # this is the name of the section used in the test, not the name of the test itself, it can be found as a parameter of this function
-    scene_name = "".join(os.path.basename(path_to_video).split(".")[:-1])
+    scene_name = orig_path_to_sections.stem
 
     movie_metadata = get_video_metadata(path_to_video)
-    section_dir_layout = get_section_dir_layout(path_to_sections)
+    section_dir_layout = get_section_dir_layout(str(path_to_sections))
     section_index = get_section_index(
-        os.path.join(path_to_sections, f"{scene_name}.json")
+        str(Path(path_to_sections).joinpath(f"{scene_name}.json"))
```

Comment:
You seem to have taken the migration a bit too seriously I suppose.

Make sure you aren't doing things like this where you make a `Path` and then immediately cast to a `str`.
Make `Path`s at places only where you can see it makes sense to do so.

Reply:
I think we might have been confused with the get_section_index function. Currently, it expects a string so we thought to convert that combined path object into a string. We plan to change the code to:

section_index = get_section_index(str(path_to_sections / f"{scene_name}.json"))

where path_to_sections is already a path object so we combine the paths then convert it into a string

If there is a better change please let us know!

Reply:
No, if we wanted strings we would've stuck with `os.path`.
The change should be that `get_section_index` function should accept a `Path` object.

Reply:
ok I can add that right now!

Reply:
I modified the get_section_index to take in a Path object instead of a string and changed the code within video_test.py to also match that.